cmd/snap/inhibit.go

```diff
@@ -0,0 +1,164 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2016 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package main
+
+import (
+	"fmt"
+	"os"
+	"os/exec"
+	"time"
+
+	"github.com/snapcore/snapd/cmd/snaplock/runinhibit"
+	"github.com/snapcore/snapd/i18n"
+)
+
+func inhibitMessage(snapName string, hint runinhibit.Hint) string {
+	switch hint {
+	case runinhibit.HintInhibitedForRefresh:
+		return fmt.Sprintf(i18n.G("snap package %q is being refreshed, please wait"), snapName)
+	default:
+		return fmt.Sprintf(i18n.G("snap package cannot be used now: %s"), string(hint))
+	}
+}
+
+func isGraphicalSession() bool {
+	return os.Getenv("DISPLAY") != "" || os.Getenv("WAYLAND_SOCKET") != ""
```

Comment:
this should also check that the `zenity` command exists, otherwise this will confuse the user I think getting an error about zenity not being found

Reply:
Good point. I'll look at what snapd userd does, as we have some of this "GUI" logic already. It's a very rough early branch for feedback so I was not going after the full tested robust implementation yet.

Reply:
I've done the minimal work required to handle missing zenity.